internal/evaluator/conftest_evaluator.go

```diff
@@ -219,7 +219,8 @@ func (c conftestEvaluator) Evaluate(ctx context.Context, inputs []string) (Check
 		exceptions := []output.Result{}
 		skipped := []output.Result{}
 
-		for _, warning := range result.Warnings {
+		for i := range result.Warnings {
```

Comment:
Why?

Reply:
Marked as a security issue, the `warning` variable is mutated for each iteration of the loop, it's address stays the same, so for each iteration a method receiving the address of such variable receives the same address. If that method was to keep the pointer and modify it, that would lead to data corruption. We don't keep the pointer in these methods so it is fairly safe, but it was still flagged by the linter.

Try [this](https://go.dev/play/p/d244Z8lyP-d) go playground code

Reply:
Oh! It's because we do this `addRuleMetadata(&warning, rules)`. That makes sense. Tricky tricky, but glad this got caught!

I suppose it doesn't affect us because `addRuleMetadata` only pull data from the warning result. We'd see a bug if we actually stored it in full.